app/src/main/java/org/oppia/app/player/content/ContentCardAdapter.kt

```diff
@@ -0,0 +1,110 @@
+package org.oppia.app.player.content
+
+import android.content.Context
+import android.text.Html
+import android.text.Spannable
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.databinding.library.baseAdapters.BR
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.content_card_items.view.*
+import org.oppia.app.R
+import org.oppia.app.databinding.ContentCardItemsBinding
+import org.oppia.app.databinding.LearnersCardItemBinding
+import org.oppia.data.backends.gae.model.GaeSubtitledHtml
+import org.oppia.util.data.UrlImageParser
+
+const val CUSTOM_TAG = "oppia-noninteractive-image"
+const val HTML_TAG = "img"
+const val CUSTOM_ATTRIBUTE = "filepath-with-value"
+const val HTML_ATTRIBUTE = "src"
+
+/** Adapter to bind the contents to the [RecyclerView]. It handles rich-text content. */
+class ContentCardAdapter(
+  private val context: Context,
+  private val entity_type: String,
+  private val entity_id: String,
+  val contentList: MutableList<GaeSubtitledHtml>
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+  private val VIEW_TYPE_CONTENT = 1
+  private val VIEW_TYPE_INTERACTION = 2
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      VIEW_TYPE_CONTENT -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<ContentCardItemsBinding>(inflater, R.layout.content_card_items, parent, false)
+        ContentViewHolder(binding)
+      }
+      VIEW_TYPE_INTERACTION -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<LearnersCardItemBinding>(inflater, R.layout.learners_card_item, parent, false)
+        LearnersViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_CONTENT -> (holder as ContentViewHolder).bind(contentList!!.get(position).html)
+      VIEW_TYPE_INTERACTION -> (holder as LearnersViewHolder).bind(contentList!!.get(position).html)
+    }
+  }
+
+  // Determines the appropriate ViewType according to the interaction type.
+  override fun getItemViewType(position: Int): Int {
+    return if (!contentList!!.get(position).contentId!!.contains("content") &&
+      !contentList!!.get(position).contentId!!.contains(
+        "Feedback")) {
+      VIEW_TYPE_INTERACTION
+    } else {
+      VIEW_TYPE_CONTENT
+    }
+  }
+
+  override fun getItemCount(): Int {
+    return contentList!!.size
+  }
+
+  inner class ContentViewHolder(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
```

Comment:
Nit here & below: please make this private since this class doesn't need to be exposed outside of the adapter.

Reply:
made inner class private